src/analysis/buildTypeWeights.ts

```diff
@@ -193,18 +220,95 @@ function parseTypes(schema: GraphQLSchema, typeWeights: TypeWeightSet): TypeWeig
                     weight: typeWeights.scalar,
                 };
             } else if (isUnionType(currentType)) {
-                // FIXME: will need information on fields inorder calculate comlpextiy
-                result[typeName] = {
-                    fields: {},
-                    weight: typeWeights.object,
-                };
+                unions.push(currentType);
             } else {
+                // FIXME: Scalar types are listed here throw new Error(`ERROR: buildTypeWeight: Unsupported type: ${currentType}`);
                 // ? what else can get through here
                 // ? inputTypes?
             }
         }
     });
 
+    unions.forEach((unionType: GraphQLUnionType) => {
```

Comment:
You could consider breaking this out into it's own function, a function that builds the common fields, and then add them to the Type Weight Object here.